test/document_stores/test_faiss.py

```diff
@@ -311,8 +284,45 @@ def test_pipeline_with_existing_faiss_docstore(tmp_path):
     nodes:
       - name: DPRRetriever
         inputs: [Query]
-    """
-    pipeline = Pipeline.load_from_config(yaml.safe_load(query_config))
-    existing_document_store = pipeline.get_document_store()
-    faiss_index = existing_document_store.faiss_indexes["document"]
-    assert faiss_index.ntotal == len(DOCUMENTS)
+"""
+        pipeline = Pipeline.load_from_config(yaml.safe_load(query_config))
```

Comment:
Instead of using `Pipeline.load_from_config(yaml.safe_load(pipeline_config))`, let's do one less step. We should rewrite the above YAML into a normal dictionary and just load it.

Untested, looks a bit like:

```python
pipeline_config = {
'version': 'ignore',
'components':[
    {
        'name': 'Retriever',
        'type': 'MockDenseRetriever',
        'params': {
             'document_store': 'ExistingFAISSDocumentStore'
         }
    },
    {
        'name': 'ExistingFAISSDocumentStore',
        'type': 'FAISSDocumentStore',
        'params': {
             'faiss_index_path': tmp_path / 'existing_faiss_document_store'
         }
    }
],
'pipelines': [
    {
        'name': 'query_pipeline',
        'nodes': {
             'name': 'Retriever',
             'inputs': ['Query']
         }
    },
]
}

pipeline = Pipeline.load_from_config(pipeline_config)
```